### PR TITLE
rpc: Fix for segfault if combinepsbt called with empty inputs

### LINKED ISSUE #15300: rpc: segfault if combinepsbt called with empty inputs
Running master 4d661ba:
```
src/bitcoin-cli combinepsbt [""] or src/bitcoin-cli combinepsbt []
```

```
  * frame #0: 0x000000010016054a bitcoind`PartiallySignedTransaction::PartiallySignedTransaction(PartiallySignedTransaction const&) [inlined] boost::optional_detail::optional_base<CMutableTransaction>::is_initialized(this=0x0000000000000000) const at optional.hpp:394 [opt]
    frame #1: 0x000000010016054a bitcoind`PartiallySignedTransaction::PartiallySignedTransaction(PartiallySignedTransaction const&) [inlined] boost::optional_detail::optional_base<CMutableTransaction>::optional_base(this=<unavailable>, rhs=0x0000000000000000) at optional.hpp:196 [opt]
    frame #2: 0x0000000100160542 bitcoind`PartiallySignedTransaction::PartiallySignedTransaction(PartiallySignedTransaction const&) [inlined] boost::optional<CMutableTransaction>::optional(this=0x000070000b9fe7c0, (null)=0x0000000000000000) at optional.hpp:958 [opt]
    frame #3: 0x0000000100160542 bitcoind`PartiallySignedTransaction::PartiallySignedTransaction(PartiallySignedTransaction const&) [inlined] boost::optional<CMutableTransaction>::optional(this=0x000070000b9fe7c0, (null)=0x0000000000000000) at optional.hpp:958 [opt]
    frame #4: 0x0000000100160542 bitcoind`PartiallySignedTransaction::PartiallySignedTransaction(this=0x000070000b9fe7c0, psbt_in=0x0000000000000000) at sign.h:576 [opt]
    frame #5: 0x0000000100142ae5 bitcoind`combinepsbt(JSONRPCRequest const&) [inlined] PartiallySignedTransaction::PartiallySignedTransaction(this=0x000070000b9fe700, psbt_in=<unavailable>) at sign.h:576 [opt]
    frame #6: 0x0000000100142ae0 bitcoind`combinepsbt(request=<unavailable>) at rawtransaction.cpp:1526 [opt]
    frame #7: 0x0000000100170f0a bitcoind`CRPCTable::execute(this=<unavailable>, request=0x000070000b9fead0) const at server.cpp:557 [opt]
```

### PR DESCRIPTION
**Pull request by benthecarman**
Fixes #15300

### CODE REVIEW

**Comment by benthecarman** [comment]
@practicalswift I did a quick pass looking for uses of get_array() and didn't find any where else that caused a crash, however some gave errors that didn't seem to correlate with what would be going wrong.

File: src/rpc/rawtransaction.cpp [hunk 1]
```diff
@@ -1514,6 +1514,9 @@ UniValue combinepsbt(const JSONRPCRequest& request)
     // Unserialize the transactions
     std::vector<PartiallySignedTransaction> psbtxs;
     UniValue txs = request.params[0].get_array();
+    if (txs.size() <= 0) {
```
Comment (promag) [inline src/rpc/rawtransaction.cpp]: `if (txs.empty()) {`

File: src/rpc/rawtransaction.cpp [hunk 2]
```diff
@@ -1514,6 +1514,9 @@ UniValue combinepsbt(const JSONRPCRequest& request)
...
+        throw JSONRPCError(RPC_INVALID_PARAMETER, "Parameter \'txs\' cannot be empty");
```
Comment (promag) [inline src/rpc/rawtransaction.cpp]: Why `\'`?

Comment (practicalswift) [reply src/rpc/rawtransaction.cpp]: `'` doesn't have to be quoted :-)

Comment (benthecarman) [reply src/rpc/rawtransaction.cpp]: TIL

**Comment by Sjors** [comment]
tACK 30d0f7b

**Pull request merged**